controllers/components/kserve/kserve_controller_actions.go

```diff
@@ -0,0 +1,267 @@
+package kserve
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	odherrors "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/errors"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+func checkPreConditions(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	k, ok := rr.Instance.(*componentsv1.Kserve)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Kserve)", rr.Instance)
+	}
+
+	if k.Spec.Serving.ManagementState == operatorv1.Managed {
```

Comment:
it is usually a good practice to keep the happy path on the left, so this may be rewritten as:

```go
if k.Spec.Serving.ManagementState != operatorv1.Managed {
    return nil
}
```

Reply:
Yeah, I was debating which path to keep on the left as "happier" :slightly_smiling_face: this guideline is more obvious to apply when one can return early with an _error_, but less so when it's not an error.

In any case, I think your suggestion is good because it removes a level of nesting, so I've added it, thanks! :+1: